torch_xla/csrc/runtime/pjrt_registry.cc

```diff
@@ -43,21 +40,18 @@ xla::GpuAllocatorConfig GetGpuAllocatorConfig() {
   return allocator_config;
 }
 
-std::optional<PluginEntry> GetPjRtPlugin(const std::string& device_type) {
+std::shared_ptr<const PjRtPlugin> GetPjRtPlugin(
```

Comment:
any particualr reason for this optional -> shared_ptr change?

Reply:
`PjRtPlugin` has to be a pointer or a reference now that we don't know the concrete type (whereas `PluginEntry` was just a value). My first thought was to make this an optional reference, but apparently [C++ doesn't support that](https://stackoverflow.com/a/26895581). An optional pointer would be unwieldy, because we'd have two layers of indirection/nullability (empty optional, and an optional holding `nullptr`). So I'm just returning `shared_ptr` here and letting `nullptr` represent the empty value.